### PR TITLE
doc: use common format for commands with sub-commands

### PR DESCRIPTION
**Pull request by garlick**
Problem: some man pages for commands with sub-commands are a mess.

Following the pattern of flux-job(1), which was channeling git-remote(1), reorganize a some section 1 pages to group sub-commands in sub-sections.

More pages to go so I'll mark as a WIP, although any feedback is welcome on what's here now.

Oh, I also tacked on a commit that flattens out the section 3 man page list.  IMHO the subtopic organization wasn't helping and it was a bit of a pain to maintain.  This makes the index glob-based like the other sections.

### CODE REVIEW

**Comment by garlick** [comment]
I completed a pass through all the commands that have subcommands, so dropping the WIP.

File: doc/man1/flux-cron.rst [hunk 1]
```diff
@@ -6,182 +6,262 @@ flux-cron(1)
...
+  Export the current environment to be used for the command being executed
+  as part of the cron job.  Normally, the broker environment is used.
 
 .. option:: -d, --working-dir=DIR
 
-      The :option:`--working-dir` option allows the working directory to be set
-      for the command being executed as part of the cron job. Normally, the
-      working directory of the broker is used.
+  Set the working directory to be set for the command being executed as part
+  of the cron job. Normally, the working directory of the broker is used.
```
Comment (grondo) [inline doc/man1/flux-cron.rst:110]: Suggestion: Drop "to be set" here so it is just "Set the working directory for the command being executed..."

Comment (grondo) [reply doc/man1/flux-cron.rst:110]: Oh looking below maybe this is just a typo and you meant "Set the working directory _to be used_..."

File: doc/man1/flux-cron.rst [hunk 2]
```diff
@@ -6,182 +6,262 @@ flux-cron(1)
...
-   Normally, the broker environment is used.
+  Export the current environment to be used for the command being executed
+  as part of the cron job.  Normally, the broker environment is used.
 
 .. option:: -d, --working-dir=DIR
 
-   The :option:`--working-dir` option allows the working directory to be set
-   for the command being executed as part of the cron job. Normally, the
-   working directory of the broker is used.
+  Set the working directory to be set for the command being executed as part
```
Comment (grondo) [inline doc/man1/flux-cron.rst:189]: s/to be set/to be used/ again

File: doc/man1/flux-overlay.rst [hunk 3]
```diff
@@ -9,42 +9,109 @@ flux-overlay(1)
...
+   'never', or 'auto' (default).
+
+.. option:: -H, --highlight=TARGET
+
+   Highlight one or more targets and their ancestors.
+
+.. option:: -w, --wait=STATE
+
+   Wait until sub-tree enters *STATE* before reporting (full, partial, offline,
+   degraded, lost)>
```
Comment (grondo) [inline doc/man1/flux-overlay.rst:80]: Possible extra trailing `>` here?

File: doc/man1/flux-overlay.rst [hunk 4]
```diff
@@ -9,42 +9,109 @@ flux-overlay(1)
...
 
-By default, the status of the full Flux instance is shown in graphical form,
-e.g.
+.. program:: flux overlay disconnect
+
+Disconnect a subtree rooted at *target* (hostname or rank).
+
+.. option:: -r, --parent=NODEID
+
+  Set parent rank to *NODEID*.   By default, the parent is determined from
```
Comment (grondo) [inline doc/man1/flux-overlay.rst:106]: very minor: extra space(s) in front of "By default, ..."

**Review by grondo — APPROVED** [review]
Great improvement, and LGTM! Just noticed some very minor things on a read-through.

File: doc/man1/flux-cron.rst [hunk 5]
```diff
@@ -6,182 +6,262 @@ flux-cron(1)
...
+For a list of keys run :option:`flux cron dump ID`.
+
+sync
+----
+
+.. program flux cron sync
+
+Query and modify the current **sync-event** behavior for the cron module.
+If a sync-event is set, the cron module will defer all task execution
+until an event matching the sync-event *topic* is received.
```
Comment (chu11) [inline doc/man1/flux-cron.rst:247]: I feel just a bit more would be useful.  Maybe mention is used for ... managing crons so they limit interruption of tasks?   sometimes used with `flux-event`?

File: doc/man1/flux-event.rst [hunk 6]
```diff
@@ -26,27 +27,56 @@ A subscription to the empty string matches all events.
...
-   followed by the payload, if any.
+pub
+---
+
+.. program:: flux event pub
+
+Publish an event on *topic* with optional *payload*. If payload is specified,
+it is interpreted as JSON unless other options are selected.  If the payload
+spans multiple arguments, the arguments are concatenated with one space
+between them.
```
Comment (chu11) [inline doc/man1/flux-event.rst:38]: perhaps include one example?

File: doc/man1/flux-getattr.rst [hunk 7]
```diff
@@ -6,28 +6,50 @@ flux-getattr(1)
...
 
-**flux** **setattr** *name* *value*
 
-**flux** **setattr** *name*
+DESCRIPTION
+===========
 
-**flux** **lsattr** [*--values*]
+Flux broker attributes provide a primitive configuration mechanism
+that predates :man1:`flux-config`.
```
Comment (chu11) [inline doc/man1/flux-getattr.rst:18]: Is "a method for the broker to export information needed by Flux services and utilities." no longer true?  I think it'd be nice to say "it's mostly used for ..." now.

Comment (chu11) [reply doc/man1/flux-getattr.rst:18]: also, i see its in the "SEE ALSO", but maybe a "see list of attributes in flux-broker-attributes(7)` might be nice.

**Review by chu11 — COMMENTED** [review]
through `flux-jobtap` changes ... taking a break, lots of reading :-)

i mentioned a few could use examples.  Maybe is for a different PR?  Could create issue for that later and just list those specific manpages.

File: doc/man1/flux-config.rst [hunk 8]
```diff
@@ -53,36 +76,25 @@ configuration key names.  This command is available to all users.
...
 
-   Require that the value has the specified type, or exit with a nonzero exit
-   code.  Valid types are *string*, *integer*, *real*, *boolean*, *object*, and
-   *array*.  In addition, types of *fsd*, *fsd-integer*, and *fsd-real* ensure
-   that a value is a both a string and valid Flux Standard Duration.
-   *fsd* prints the value in its human-readable, string form. *fsd-integer*
-   and *fsd-real* print the value in integer and real seconds, respectively.
+.. program:: flux config reload
 
+:program:`flux config reload` tells :man1:`flux-broker` to reload its TOML
```
[the remark this answers is not in the document]
Comment (garlick) [reply doc/man1/flux-config.rst:92]: Great suggestions!  Starting a new branch so these can go in a new PR as suggested.

**Comment by garlick** [comment]
pushed with fixes to @grondo's comments.

File: doc/man1/flux-cron.rst [hunk 9]
```diff
@@ -6,182 +6,262 @@ flux-cron(1)
...
-   seconds, *m* for minutes, *h* for hours and *d* for days.
-   Options:
+The following sub-commands are available:
 
-.. program:: flux cron interval
+tab
+---
 
-.. option:: -n, --name=STRING
+.. program:: flux cron tab
```
[the remark this answers is not in the document]
Comment (garlick) [reply doc/man1/flux-cron.rst:50]: Deferring the flux-cron suggestions to an issue:  #5600

File: doc/man1/flux-getattr.rst (hunk 7 above)
Comment (garlick) [reply doc/man1/flux-getattr.rst]: Since this was really about text that I changed in this PR, I'll go ahead and tack on a fix for this here.

**Comment by garlick** [comment]
I'm going to go ahead and set MWP here - hopefully that's OK!

**Review by chu11 — COMMENTED** [review]
just went through the rest of the commits, nothing else caught my eye

**Pull request merged**